pkg/acs/calacs/lib/prinfo.c

```diff
@@ -292,15 +292,15 @@ int maxch     i: allocated length of outstr (including EOS)
    to lower case, and pad with blanks.
 */
 
-void Upper1 (char *instr, char *outstr, int maxch) {
+void Upper1 (const char *instr, char *outstr, int maxch) {
 
 /* arguments:
 char *instr   i: input string
 char *outstr  o: string with only the first letter capitalized
 int maxch     i: allocated length of outstr (including EOS)
 */
 
-	int i;
+	int i = 1;
```

Comment:
IMHO the original code just never accounted for null string case. If I have my way, I would initialize `i = 0` *and* add your logic above. Your current fix only gets rid of the compilation warning but does not really address the logic for null string.